public/pages/DetectorResults/containers/AnomalyHistory.tsx

```diff
@@ -66,16 +77,78 @@ export const AnomalyHistory = (props: AnomalyHistoryProps) => {
     ANOMALY_HISTORY_TABS.FEATURE_BREAKDOWN
   );
 
+  const [isLoadingAnomalyResults, setIsLoadingAnomalyResults] = useState<
+    boolean
+  >(false);
+  const [bucketizedAnomalyResults, setBucketizedAnomalyResults] = useState<
+    Anomalies
+  >();
+  const [pureAnomalies, setPureAnomalies] = useState<AnomalyData[]>([]);
+  const [bucketizedAnomalySummary, setBucketizedAnomalySummary] = useState<
+    AnomalySummary
+  >(INITIAL_ANOMALY_SUMMARY);
+
   useEffect(() => {
-    getAnomalyResultsWithDateRange(
-      dispatch,
-      dateRange.startDate,
-      dateRange.endDate,
-      props.detector.id
-    );
+    async function getAnomalyResults() {
+      try {
+        setIsLoadingAnomalyResults(true);
+        const anomalySummaryResult = await dispatch(
+          searchES(
+            getAnomalySummaryQuery(
+              dateRange.startDate,
+              dateRange.endDate,
+              props.detector.id
+            )
+          )
+        );
+        setPureAnomalies(parsePureAnomalies(anomalySummaryResult));
+        setBucketizedAnomalySummary(parseAnomalySummary(anomalySummaryResult));
+        const result = await dispatch(
+          searchES(
+            getAnomalyResultsQuery(
+              dateRange.startDate,
+              dateRange.endDate,
+              1,
+              props.detector.id
+            )
+          )
+        );
+        setBucketizedAnomalyResults(parseAnomalyResults(result));
+        setIsLoadingAnomalyResults(false);
+      } catch (err) {
+        setIsLoadingAnomalyResults(false);
```

Comment:
just move `setIsLoadingAnomalyResults(false)` to outside of `try catch` or `finally`

Reply:
Good point, will put in finally